fuzzers/main.py

```diff
@@ -0,0 +1,969 @@
+                            str(fuzzer.log_fullpath))
+                    fuzzer.report_crash(corpus, bucket)
+                    fuzzers.remove(fuzzer)
+
+                    # Start a new fuzzer
+                    fuzzer = configure_one_fuzzer(repo, corpus, sync_log_files,
+                                                  fuzzers)
+                    if pause_exit_spot(pause_evt, resume_evt, exit_evt):
+                        return
+                    fuzzer.build()
+                    if pause_exit_spot(pause_evt, resume_evt, exit_evt):
+                        return
+                    fuzzer.start(corpus)
+                    if pause_exit_spot(pause_evt, resume_evt, exit_evt):
+                        return
+
+            # Regularly upload the sync log files
+            upload_interval_secs = SYNC_LOG_UPLOAD_INTERVAL.total_seconds()
+            next_sync = last_sync_file_upload + upload_interval_secs
+            if time.monotonic() > next_sync:
+                last_sync_file_upload = time.monotonic()
+                for line in sync_log_files:
+                    bucket.blob(f'logs/{line}').upload_from_filename(
+                        str(LOGS_DIR / line))
+
+    # TODO: Minimize the corpus
+    # TODO: Rsync the corpus from gcs more frequently, not just once per fuzzer restart
+    # TODO: Add corpus size gauge metric
+    # TODO: Add coverage metrics (will need to parse logs?)
+    # TODO: Add metrics about overhead (time to update corpus, to build fuzzer, etc.)
+
+
+def listen_for_commands(pause_event: threading.Event,
+                        resume_event: threading.Event) -> None:
+    """
+    Spawn an HTTP server to remote control the fuzzers
+
+    `/pause` and `/resume` respectively sigstop and sigcont the fuzzers
+
+    Args:
+        pause_event: the pausing event to trigger upon receiving a request on /pause
+        resume_event: the resuming event to trigger upon receiving a request on /resume
+    """
+
+    class HTTPHandler(BaseHTTPRequestHandler):
+
+        def do_GET(self) -> None:  # pylint: disable=invalid-name
+            if self.path == '/pause':
+                resume_event.clear()
+                pause_event.set()
+                self.send_response(200)
+            elif self.path == '/resume':
+                pause_event.clear()
+                resume_event.set()
+                self.send_response(200)
+            else:
+                self.send_response(404)
+            self.end_headers()
+
+    with HTTPServer(('127.0.0.1', CMD_PORT), HTTPHandler) as httpd:
+        print(f'Serving command server on port {CMD_PORT}', file=sys.stderr)
+        httpd.serve_forever()
+
+
+THREAD_EXCEPTION: typing.Optional[threading.ExceptHookArgs] = None
+EXCEPTION_HAPPENED_IN_THREAD = threading.Event()
+
+
+def main() -> None:
+    """Main function"""
+
+    # Make sure to cleanup upon ctrl-c or upon any exception in a thread
+    def new_excepthook(args: typing.Any) -> None:
```

Comment:
This should be `args: threading.ExceptHookArgs` it seems.

Reply:
Done, though it doesn't actually fix the mypy issue